### PR TITLE
CLOUDP-199913 Add watch (-w) flag to atlas deployments search index create command

### PR DESCRIPTION
**Pull request by fmenezes**
## Proposed changes
Add watch (-w) flag to atlas deployments search index create command

_Jira ticket:_ CLOUDP-199913

Closes #[issue number]

## Checklist

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [ ] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [ ] I have updated [test/README.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/README.md) (if an e2e test has been added)
- [ ] I have run `make fmt` and formatted my code

### CODE REVIEW

File: internal/cli/atlas/deployments/search/indexes/create.go [hunk 1]
```diff
@@ -80,6 +85,62 @@ func (opts *CreateOpts) initMongoDBClient(ctx context.Context) func() error {
...
+		},
+	})
+	if err != nil {
+		return "", err
+	}
+	var results []bson.M
+	err = cursor.All(ctx, &results)
+	if err != nil {
+		return "", err
+	}
```
Comment (andreaangiolillo) [inline internal/cli/atlas/deployments/search/indexes/create.go:108]:
```suggestion
	if err = cursor.All(ctx, &results); err != nil {
		return "", err
	}
```

File: test/e2e/atlas/deployments_local_test.go [hunk 2]
```diff
@@ -186,33 +184,35 @@ test   LOCAL   7.0.1     IDLE
...
-			if status == "STEADY" {
-				indexID, _ = results[0]["id"].(string)
-				break
-			}
-		}
+
+	t.Run("Index List", func(t *testing.T) {
+		cmd := exec.Command(cliPath,
+			deploymentEntity,
+			searchEntity,
```
Comment (andreaangiolillo) [inline test/e2e/atlas/deployments_local_test.go:195]: Why do we need this? we are testing index list at line 254. Can we remove it?

**Review by andreaangiolillo — DISMISSED** [review]
LGTM

File: test/e2e/atlas/deployments_local_test.go (hunk 2 above)
Comment (fmenezes) [reply test/e2e/atlas/deployments_local_test.go:195]: I had to move test sooner, deleted the duplication, it was a bad merge

**Review by andreaangiolillo — APPROVED** [review]

**Pull request merged**